webapp/app/forms/steps/steuerlotse_step.py

```diff
@@ -0,0 +1,159 @@
+        redirection = self._handle_redirects()
+        if redirection:
+            return redirection
+        return self.render()
+
+    def _main_handle(self, stored_data):
+        return stored_data
+
+    def render(self):
+        raise NotImplementedError
+
+    def _get_session_data(self, ttl: Optional[int] = None):
+        serialized_session = session.get('form_data', b"")
+
+        if self.default_data:
+            stored_data = self.default_data | deserialize_session_data(serialized_session, ttl)  # updates session_data only with non_existent values
+        else:
+            stored_data = deserialize_session_data(serialized_session, ttl)
+
+        return stored_data
+
+    def url_for_step(self, step_name, _has_link_overview=None, **values):
+        """Generate URL for given step and current session."""
+        if not _has_link_overview:
+            _has_link_overview = self.has_link_overview
+
+        # show overview buttons if explicitly requested or if shown for current request
+        return url_for(self.endpoint,
+                       step=step_name,
+                       link_overview=_has_link_overview,
+                       **values)
+
+
+class FormSteuerlotseStep(SteuerlotseStep):
+
+    def __init__(self, title, form, intro=None, endpoint=None, overview_step=None, default_data=None, prev_step=None, next_step=None, header_title=None,
+                 template='basis/form_full_width.html'):
+        super(FormSteuerlotseStep, self).__init__(title, intro, endpoint, overview_step, default_data, prev_step, next_step, header_title)
+        self.form = form
+        self.template = template
+
+    def _pre_handle(self, stored_data):
+        super()._pre_handle(stored_data)
+        form = self.create_form(request, prefilled_data=stored_data)
+        if request.method == 'POST' and form.validate():
+            stored_data.update(form.data)
+        self.render_info.form = form
+        return stored_data
+
+    def _post_handle(self, stored_data):
+        override_session_data(stored_data)
+
+        redirection = self._handle_redirects()
+        if redirection:
+            return redirection
+
+        if request.method == 'POST' and self.render_info.form.validate():
+            return redirect(self.render_info.next_url)
+        return self.render()
+
+    def create_form(self, request, prefilled_data):
+        # If `form_data` is present it will always override `data` during
+        # value binding. For `BooleanFields` an empty/missing value in the `form_data`
+        # will lead to an unchecked box.
+        form_data = request.form
+        if len(form_data) == 0:
+            form_data = None
+
+        form = self.form(form_data, **prefilled_data)
+        return form
+
+    def render(self):
+        """
+        Renders a Form step. Use the render_info to provide all the needed data for rendering.
+        """
+        self.render_info.form.first_field = next(iter(self.render_info.form))
```

Comment:
I think that this might be the reason.
I had a long search in our archive where this is from and this has been here since the beginning. As we do not use it in the templates we could a) either delete it or b) use it in the templates to set autofocus on the first field.

If we want to do b), I would still do a), create a ticket and write this line in the comment section.